users/userRouter.js

```diff
@@ -1,47 +1,173 @@
+            if (posts) {
+                res.status(200).json(posts)
+            } else {
+                res.status(999).json({ error: "Another horrible USER error"})
+            }
+        } catch {
+            res.status(500).json({ error: "oops../:id/posts be broken"})
+        }
 });
-
-//custom middleware
-
-function validateUserId(req, res, next) {
-
+// POST
+    router.post('/', async (req, res) => {
+        const user = req.body
+        console.log(user)
+        try {
+            const newUser = await db.insert(user)
+            if (newUser) {
+                res.status(200).json(newUser)
+            } else {
+                res.status(500).json({ error: "bro we sorry"})
+            }
+        } catch {
+            res
+                .status(500)
+                .json({ error: 'The Post information could not be retrieved'})
+        }
+
+
+    });
+
+    router.post('/:id/posts', validateUserId, async (req, res) => {
+        const { id } = req.params
+        try {
+            const newPost = await db.getUserPosts(id)
+
+        } catch {
+            res
+                .status(500)
+                .json({ error: 'The Post information could not be retrieved'})
+        }
+
+    });
+// DELETE
+    router.delete('/:id', validateUserId, async (req, res) => {
+        const { id } = req.params
+        try {
+            const deletedUser = await db.remove(id)
+            if (deletedUser) {
+                res.status(200).json(deletedUser)
+            } else {
+                res.status(999).json({ error: "user could not be deleted"})
+            }
+        } catch {
+            res.status(500).json({ error: "user could not be deleted"})
+        }
+
+
+    });
+// PUT
+    router.put('/:id', validateUserId, validateUser, async (req, res) => {
+        // const { id } = req.params
+        // const { name } = req.body
+        // console.log("name", name)
+        // console.log('id', id)
+
+        try {
+            if (name) {
+                const newName = await db.update(id, req.body)
+
+                if (newName) {
+                    res.status(200).json(newName)
+                } else {
+                    res
+                        .status(404)
+                        .json({message: "prepared object broken"})
+                }
+            } else {
+                res.status(500).json({ error: "The user could not be edited"})
+            }
+        } catch {
+            res.status(500).json({ error: "The user could not be edited"})
+        }
+    });
+
+
+
+/// CUSTOM MIDDLEWARE
+async function validateUserId(req, res, next) {
+    console.log('validateUserId Middleware')
+    const { id } = req.params;
+
+    const user = await db.getById(id)
+    
+    if (user) {
+        req.user = user;
+        next();
+    } else {
+        res.status(404).json({ message: 'User not found; invalid ID'})
+    }
 };
 
 function validateUser(req, res, next) {
```

Comment:
I noticed that this MW is not being used in your post route. I did some refactoring to make use of this custom MW. Refer to another code suggestion in your user post route. Basically, all this is doing is checking if the name field is not an empty string. If the field is not empty it moves on to your post route's try/catch using `next()`.
```suggestion
function validateUser(req, res, next) {
  const bodyToPass = req.body;
  if (bodyToPass.name) {
    next();
  } else {
    res.status(400).json({message: "missing required name field"});
  }
}
```